crypto/ecdsa.go

```diff
@@ -67,6 +67,15 @@ func ECDSAKeyPairFromKey(priv *ecdsa.PrivateKey) (PrivKey, PubKey, error) {
 	return &ECDSAPrivateKey{priv}, &ECDSAPublicKey{&priv.PublicKey}, nil
 }
 
+// ECDSAPublicKeyFromPubKey generates a new ecdsa public key from an input public key
+func ECDSAPublicKeyFromPubKey(pub *ecdsa.PublicKey) (PubKey, error) {
+	if pub == nil {
+		return nil, ErrNilPublicKey
+	}
+
+	return &ECDSAPublicKey{pub}, nil
+}
```

Comment:
For context, the bug I see here is that someone could pass a key into it and since it's by reference modify it while libp2p is using it (libp2p expects key to be constant and not change).

If it's passed by value then a copy is made and this can't happen.
```suggestion
func ECDSAPublicKeyFromPubKey(pub ecdsa.PublicKey) (PubKey, error) {
  return &ECDSAPublicKey{pub:&pub}, nil
}
```